src/vs/base/parts/storage/common/storage.ts

```diff
@@ -224,7 +239,7 @@ export class Storage extends Disposable implements IStorage {
 		}
 
 		// Otherwise, convert to String and store
-		const valueStr = String(value);
+		const valueStr = typeof value === 'object' && value !== null ? JSON.stringify(value) : String(value);
```

Comment:
I wonder if we should use `types.isObject` instead, otherwise you would match on `Date` and `RegExp` as well which we probably do not want?

Reply:
Also, why do we need again to check for `value` being `null`?